src/tools/copyset_check_core.cpp

```diff
@@ -351,9 +357,31 @@ int CopysetCheckCore::CheckCopysetsOnServer(const std::string& serverIp,
     return CheckCopysetsOnServer(0, serverIp, true, unhealthyChunkServers);
 }
 
+void CopysetCheckCore::ConcurrentCheckCopysetsOnServer(
+                const std::vector<ChunkServerInfo> &chunkservers,
+                uint32_t *index, std::map<std::string,
+                std::pair<int, std::shared_ptr<butil::IOBuf>>> *result) {
+    while (1) {
+        indexMutex.lock();
+        if (*index + 1 > chunkservers.size()) {
+            indexMutex.unlock();
+            break;
+        }
+        auto info = chunkservers[*index];
+        (*index)++;
+        indexMutex.unlock();
+        std::string csAddr = info.hostip() + ":" + std::to_string(info.port());
+        std::shared_ptr<butil::IOBuf> iobuf = std::make_shared<butil::IOBuf>();
```

Comment:
IOBuf just holds some pointers which store real data address, it cheep to copy it.
so, it's not necessary to use shared_ptr to wrap it.

Reply:
> IOBuf just holds some pointers which store real data address, it cheep to copy it.
> so, it's not necessary to use shared_ptr to wrap it.

fixed